modules/publication/ajax/FileUpload.php

```diff
@@ -0,0 +1,877 @@
+    $desc            = $_POST['description'] ?? null;
+    $leadInvest      = $_POST['leadInvestigator'] ?? null;
+    $leadInvestEmail = $_POST['leadInvestigatorEmail'] ?? null;
+
+    // check if lead investigator already exists in collaborator table
+    // use ID if exists, else insert
+    $leadInvID = $db->pselectOne(
+        'SELECT PublicationCollaboratorID '.
+        'FROM publication_collaborator '.
+        'WHERE Name = :n OR Email = :e',
+        array(
+         'n' => $leadInvest,
+         'e' => $leadInvestEmail,
+        )
+    );
+    if (empty($leadInvID)) {
+        $db->insert(
+            'publication_collaborator',
+            array(
+             'Name'  => $leadInvest,
+             'Email' => $leadInvestEmail,
+            )
+        );
+
+        $leadInvID = $db->getLastInsertId();
+    }
+    if (!isset($desc, $leadInvest, $leadInvestEmail)) {
+        showPublicationError('A mandatory field is missing!', 400);
+    }
+    // INSERT INTO publication ...
+    $uid   = $user->getId();
+    $today = date('Y-m-d');
+    // insert the titleRaw to avoid double escaping
+    $fields = array(
+               'UserID'             => $uid,
+               'Title'              => $titleRaw,
+               'Description'        => $desc,
+               'LeadInvestigatorID' => $leadInvID,
+               'DateProposed'       => $today,
+              );
+
+    $db->insert('publication', $fields);
+    $pubID = $db->getLastInsertId();
+
+    try {
+        // process files
+        processFiles($pubID);
+        // INSERT INTO publication_collaborator
+        insertCollaborators($pubID);
+        // INSERT INTO publication_users_edit_perm_rel
+        insertEditors($pubID);
+        // INSERT INTO publication_keyword
+        insertKeywords($pubID);
+        // INSERT INTO publication_parameter_type_rel
+        insertVOIs($pubID);
+    } catch (Exception $e) {
+        cleanup($pubID);
+        showPublicationError($e->getMessage(), 500);
+    }
+
+    notify($pubID, 'submission');
+}
+
+/**
+ * Function for storing files and inserting file meta data into database
+ *
+ * @param int $pubID ID of Publication in publication table
+ *
+ * @return void
+ */
+function processFiles($pubID) : void
+{
+    if (empty($_FILES)) {
+        return;
+    }
+    $db     = \Database::singleton();
+    $config = \NDB_Config::singleton();
+
+    $publicationPath = $config->getSetting('publication_uploads');
+
+    if (!is_dir($publicationPath)) {
+        throw new ConfigurationException(
+            "Error! The upload folder '$publicationPath' does not exist!"
+        );
+    }
+
+    foreach ($_FILES as $name => $values) {
+        $fileName = preg_replace('/\s/', '_', $values["name"]);
+        if (file_exists($publicationPath . $fileName)) {
+            throw new LorisException("File $fileName already exists!");
+        }
+        $extension = pathinfo($fileName)['extension'];
+        $index     = preg_split('/_/', $name)[1];
```

Comment:
This is confusing.. what's special about the second piece of the filename after the first whitespace (I think? whitespace is replaced by _ above.)